packages/peregrine/lib/talons/Navigation/useNavigation.js

```diff
@@ -11,16 +12,20 @@ const ancestors = {
     MENU: null
 };
 
-export const useNavigation = () => {
+export const useNavigation = props => {
+    const { customerQuery } = props;
+
     // retrieve app state from context
     const [appState, { closeDrawer }] = useAppContext();
     const [catalogState, { actions: catalogActions }] = useCatalogContext();
-    const [, { getUserDetails }] = useUserContext();
+    const [, { actions: userActions }] = useUserContext();
+    const { data } = useQuery(customerQuery);
 
-    // request data from server
     useEffect(() => {
-        getUserDetails();
-    }, [getUserDetails]);
+        if (data) {
+            userActions.getDetails.receive(data.customer);
```

Comment:
@jimbo 
This feels wrong to me, and @sirugh and I weren't sure which direction this should go.

1. Continue storing `currentUser` state in redux. We'd need to create a `setCurrentUser` action that simulates `actions.getDetails.(request|receive)`. This would be more backward compatible, but simulating a network request makes `isGettingDetails` useless.
2. Pull `currentUser` state in `UserContext` and remove a ton of redux code. If I were writing this from scratch today, I think this is where I'd put it, so just looking for arguments against that.

This problem is challenging me because query state lives in the talons, so there's really no use for it in redux anymore. Is there a better pattern we can use here? Is there a tertiary solution here where talons pass queries off the redux that hasn't been established yet?

Reply:
>  Is there a tertiary solution here where talons pass queries off to redux that hasn't been established yet?

If we could figure out a way to do this the work to convert to graphql gets a lot simpler. We would be able to replace the async REST calls with graphql requests. Right now we're fighting two battles -- replacing rest with graphql and figuring out how to refactor from redux actions to components.

Reply:
Alternatively, in this specific case we may not need to store user data in the store anymore. I was suggesting in our chat that we just have the components fetch the data they need to render through graphql. This would mean that other components who care about `currentUser` would also have to make a `customerQuery`.

That only really works if we can inform those components of some upstream change though -- `isSignedIn` is a good example of a global state we can use to trigger updates within the components.

So that would look like this:

```js
const useNavigation = () => {
  const [{ isSignedIn }] = useUserContext();
  const [getUserDetails, { error, loading, data }] = useLazyQuery(customerQuery)
  
  useEffect(() => {
    if(isSignedIn) {
      getUserDetails()
    }
  }, [...]

  if (data) {
    // render Navigation with user data
  }
}
```

Reply:
Good questions.

- There's a reason we make sync actions, such as `userActions.getDetails.receive()`, available from `UserContext`: sync actions are the simplest way to update the store. So this doesn't really feel wrong to me, and I don't think a `setCurrentUser` async action would add any value.

- The state we keep in the store is all global state, and there is no state more global than `currentUser`. Here, you've suggested "pulling" `currentUser` state into `UserContext`; how would you keep it? Would the `user` slice remain, and in what capacity?

- One of the reasons we chose to keep the store for global state is that our most orchestrated async actions—those in cart and checkout—rely on having access to the latest state during each step, via `getState()`. For example, in checkout, `getShippingMethods()` pulls the latest value of `user` directly from the store before determining which endpoint to use. If this state isn't in the store, then a component would have to read it from context and pass it as an argument (payload). The `user` state is unlikely to be stale, but in other cases, such as `cartId`, state received from the component could be stale by the time the async action used it.

---

Long story short, I think it's best to keep our global state in the store for now.

Reply:
- Point 1: Awesome! I had not seen any direct usage of sync actions, so good to know. This does pose a followup question below.
- Point 2: No real reason for this suggestion. `currentUser` state seemed half-baked in redux, and most components I found were triggering their own fetch before using it (also the object shape was `whatever is returned by REST`). Was really just trying to kick off the conversation about when to roll our own state into context, or use redux. Happy to leave it where it is, and try to clean up its usage.
- Point 3: Noted. I was a little narrowed on `user.currentUser`, but can still see where sharing that state might be needed to pre-fill address information, or something like that.

New Question: Do we continue adhering to the existing action map pattern? Piggy backing on @sirugh example:

```javascript
const useNavigation = () => {
  const [{ isSignedIn }, { actions: userActions }] = useUserContext();
  const [getUserDetails, { error, loading, data }] = useLazyQuery(customerQuery)
  
  useEffect(() => {
    if(isSignedIn) {
      userActions.getDetails.request(); // trigger request action
      getUserDetails()
    }
  }, [...]

  if (data || error) {
    userActions.getDetails.receive(data, error);
  }
}
```

Reply:
Do you need to use `getDetails.request`? The request sync actions are mostly used to set some global "loading data" state.

Reply:
That's pretty much the question I was asking; do we continue with this pattern of storing fetching state globally? It feels like the answer is no, but that likely means we need to cleanup our action maps to remove request, and remove fetching states from `initialState` where needed.